src/metatensor/models/experimental/gap/trainer.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import List, Union
+
+import metatensor.torch
+import torch
+from metatensor.torch import TensorMap
+
+import metatensor
+from metatensor.models.utils.data import Dataset
+
+from ...utils.composition import calculate_composition_weights
+from ...utils.data import check_datasets
+from . import GAP
+from .model import torch_tensor_map_to_core
+
+
+logger = logging.getLogger(__name__)
+
+
+class Trainer:
```

Comment:
I'm not sure I follow what's going on here. This seems to already be using #185 which is not merged yet.